llvm/include/llvm/ADT/APDecimalFloat.h

```diff
@@ -0,0 +1,260 @@
+//===- APDecimalFloat.h - Decimal Float constant handling -----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Defines the decimal float number interface.
+/// This is a class for abstracting various operations performed on decimal float 
+/// types.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_ADT_APDECIMALFLOAT_H
+#define LLVM_ADT_APDECIMALFLOAT_H
+
+#include "/Users/shafik/Downloads/IntelRDFPMathLib20U2/LIBRARY/src/bid_conf.h"
+#include "/Users/shafik/Downloads/IntelRDFPMathLib20U2/LIBRARY/src/bid_functions.h"
+#include "/Users/shafik/Downloads/IntelRDFPMathLib20U2/LIBRARY/src/bid_internal.h"
+
+#include "llvm/ADT/APFloat.h"
+#include "llvm/ADT/APSInt.h"
+#include "llvm/ADT/FoldingSet.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace llvm {
+
+class APFloat;
+class APDecimalFloat;
+struct fltSemantics;
+
+struct decFltSemantics {
+public:
+  decFltSemantics(unsigned Width, unsigned Scale, bool IsSigned,
+                  bool IsSaturated, bool HasUnsignedPadding)
+      : Width(Width), LsbWeight(-static_cast<int>(Scale)), IsSigned(IsSigned),
+        IsSaturated(IsSaturated), HasUnsignedPadding(HasUnsignedPadding) {}
```

Comment:
I think we can omit the saturation properties; we don't need to worry about saturating and non-saturating types like fixed-point does.